### ISSUE TITLE
Android Memories app login issues

### ISSUE THREAD

**Issue by kabaga** [pulsejet/memories#963]
**Describe the bug**
Downloaded the latest of the Memories Android app from the Google Playstore. The issue that I am having is I could not login to the app. When I entered my username and app token. The mobile browser shows "Account connected. your client should now be connected! You can close this window." 

However, when I closed the window, the Memories app is stuck in "Waiting for login to complete. Keep this page open in the background."

**To Reproduce**
1. Enter the URL
2. Enter the login crentials
3. Close mobile browser window
4. Go to Memories app

**Screenshots**
N/A

**Platform:**

- OS: Samsung OneUI 5.1
- Browser: Samsung browser
- Memories Version: 6.1.5
- Nextcloud Version: 27.1.4
- PHP Version: 8.2

**Additional context**

- Any errors in the JS console?
- Any errors in the Nextcloud server logs?

**Comment by pulsejet** [comment]
> Any errors in the Nextcloud server logs?

?

Also try navigating your browser to `https://your-nextcloud/apps/memories/api/describe` and check if the protocol in the output is correct (http vs https)

**Comment by kabaga** [comment]
> > Any errors in the Nextcloud server logs?
> 
> ?
> 
> Also try navigating your browser to `https://your-nextcloud/apps/memories/api/describe` and check if the protocol in the output is correct (http vs https)

I got https with that link.
About the logs, the only error log that I could see is below, which I believe is not related to the Memories.
```
OC\User\NoUserException: Backends provided no user object
```

**Comment by pulsejet** [comment]
I assume this happens on all mobile devices? That would point to a server / reverse proxy config issue. Regardless, this would need more debugging on the client side (need to logcat from the mobile)

**Comment by kabaga** [comment]
Yes, all mobile devices. I am using the OPNsense NGINX plugins as my reverse proxy. Is there an NGINX specific settings that would allow the Memories app to work?
Which logcat app should I download to get the debug information needed?

**Comment by rednus** [comment]
Same issue here.. For me, the login page reappears again and again.. Even tried to use App Passcode..

**Comment by rednus** [comment]
> Same issue here.. For me, the login page reappears again and again.. Even tried to use App Passcode..

This issue is now resolved for me by using the OVERWRITECLIURL in the compose file (I am using NX AIO)

**Comment by kabaga** [comment]
> > Same issue here.. For me, the login page reappears again and again.. Even tried to use App Passcode..
> 
> This issue is now resolved for me by using the OVERWRITECLIURL in the compose file (I am using NX AIO)
> 
> 

How would you do this on a bare metal install (VM)?

**Comment by rednus** [comment]
Assume you edit your config/config.php to include <b>overwrite.cli.url</b>

```
'trusted_domains' =>
  array (
    0 => 'localhost',
    1 => 'nextcloud.yourdomain.com',
  ),
  'datadirectory' => '/path/to/data',
  'dbtype' => 'pgsql',
  'version' => '27.1.5.1',
  'overwritehost' => 'nextcloud.yourdomain.com',
  'overwriteprotocol' => 'https',
  'overwrite.cli.url' => 'https://nextcloud.yourdomain.com/',
  'dbname' => 'nextcloud_database',
  'dbhost' => 'nextcloud-aio-database',
  'dbport' => '',
  'dbtableprefix' => 'oc_',
...
```
